src/frontend/flight/sync.rs

```diff
@@ -371,42 +400,170 @@ impl SeafowlDataSyncManager {
         )
     }
 
-    fn plan_data_sync(
+    // Get a list of files that will be scanned and removed based on the pruning qualifier.
+    fn get_removes(
```

Comment:
Basically, all these partitions are the ones that we're going to overwrite, so we'll delete these ones and then add our merged ones?

Reply:
That is correct, the same as when planning a DELETE/UPDATE.